### PR TITLE
Add gateway location RPC

### PR DESCRIPTION
**Pull request by macpie**

### CODE REVIEW

File: src/service/iot_config.proto [hunk 1]
```diff
@@ -348,6 +348,14 @@ message load_region_req_v1 {
 
 message load_region_res_v1 {}
 
+message gateway_location_req_v1 {
+  bytes gateway = 1;
+  uint64 timestamp = 2;
```
Comment (madninja) [inline src/service/iot_config.proto:353]: Why timestamp ? This is not a mutating call

File: src/service/iot_config.proto [hunk 2]
```diff
@@ -348,6 +348,14 @@ message load_region_req_v1 {
...
+  bytes signature = 3;
```
Comment (madninja) [inline src/service/iot_config.proto:354]: Who is signing this and how would you know who’s signing?

File: src/service/iot_config.proto [hunk 3]
```diff
@@ -348,6 +348,14 @@ message load_region_req_v1 {
...
+}
+
+message gateway_location_res_v1 { string location = 1; }
+
```
Comment (madninja) [inline src/service/iot_config.proto:358]: Should the config service  sign the response ? The caller should already have the signing pubkey

File: src/service/iot_config.proto (hunk 2 above)
Comment (macpie) [reply src/service/iot_config.proto:354]: This is the same as the other messages only a handful of admin keys have access to this RPC

File: src/service/iot_config.proto (hunk 3 above)
Comment (macpie) [reply src/service/iot_config.proto:358]: We don't in any other message

File: src/service/iot_config.proto (hunk 1 above)
Comment (macpie) [reply src/service/iot_config.proto:353]: avoid replays

Comment (jeffgrunewald) [reply src/service/iot_config.proto:353]: this is just an informational "getter" correct? is there a replay attack vector with that?

File: src/service/iot_config.proto (hunk 3 above)
Comment (jeffgrunewald) [reply src/service/iot_config.proto:358]: as i recall we had the config service sign responses to the gateways b/c we couldn't guarantee they were able to travel over a tls connection but in the case of the other network (router, hpr, etc) infrastructure we should be able to enforce a tls connection to the config service so the signature wasn't critical

File: src/service/iot_config.proto (hunk 2 above)
Comment (madninja) [reply src/service/iot_config.proto:354]: perhaps add a comment here so it's not confusing next time someone comes across this?

File: src/service/iot_config.proto (hunk 1 above)
Comment (macpie) [reply src/service/iot_config.proto:353]: well we just don't want anybody to hit the API? otherwise I don't care that much

Comment (macpie) [reply src/service/iot_config.proto:353]: Will remove

**Review by jeffgrunewald — APPROVED** [review]

**Pull request merged**